src/traffic_server/InkAPI.cc

```diff
@@ -6563,6 +6563,15 @@ TSVConnSslCurveGet(TSVConn sslp)
   return ssl_vc ? ssl_vc->getSSLCurve() : nullptr;
 }
 
+const char *
+TSVConnSslServerNameGet(TSVConn sslp)
+{
+  NetVConnection *vc        = reinterpret_cast<NetVConnection *>(sslp);
+  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(vc);
```

Comment:
We don't need this cast since `get_server_name` is defined in NetVConnection, and UnixNetVC returns `nullptr`. QUICNetVC currently returns `nullptr` and will return a server name.